app.js

```diff
@@ -54,8 +53,19 @@ const ProductList = React.createClass ({
     this.updateState();
     console.log(productId + " was downvoted.");
   },
+  getSortedData: function () {
+    if (this.state.sorting === 'descending') {
+        return Data.sort((a, b) => {
+            return b.votes - a.votes;
+        });
+    } else if (this.state.sorting === 'ascending') {
+        return Data.sort((b, a) => {
+            return b.votes - a.votes;
+        });
+    }
+  },
   render: function () {
-    const products = this.state.products.map((product) => {
+    const products = this.getSortedData().map((product) => {
```

Comment:
@Xerxes25x 

This isn't 100% ideal, however it solved the initial issue where the sorting didn't apply until data was changed or if you continually clicked the toggle button. 

For consideration, this may be better handled by settings the component state via lifecycle methods such as `componentWillMount` or `componentDidMount`. https://facebook.github.io/react/docs/component-specs.html